infrastructure/src/test/java/org/corfudb/infrastructure/LogUnitServerCacheTest.java

```diff
@@ -0,0 +1,77 @@
+package org.corfudb.infrastructure;
+
+import io.micrometer.core.instrument.FunctionCounter;
+import org.corfudb.common.metrics.micrometer.MeterRegistryProvider.MeterRegistryInitializer;
+import org.corfudb.infrastructure.BatchProcessor.BatchProcessorContext;
+import org.corfudb.infrastructure.LogUnitServer.LogUnitServerConfig;
+import org.corfudb.infrastructure.log.InMemoryStreamLog;
+import org.corfudb.infrastructure.log.StreamLog;
+import org.corfudb.protocols.wireprotocol.DataType;
+import org.corfudb.protocols.wireprotocol.LogData;
+import org.corfudb.util.LambdaUtils.BiOptional;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.Duration;
+import java.util.Random;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+
+class LogUnitServerCacheTest {
+    private final Random rnd = new Random();
+
+    @Test
+    public void testMetrics() {
+        String defaultMetricsLoggerName = LogUnitServerCacheTest.class.getName();
+        Duration defaultMetricsLoggingInterval = Duration.ofMinutes(1);
+
+        Logger logger = LoggerFactory.getLogger(defaultMetricsLoggerName);
+        String endpoint = "127.0.0.1:9000";
+
+        MeterRegistryInitializer.initServerMetrics(logger, defaultMetricsLoggingInterval, endpoint);
+
+        LogUnitServerConfig config = LogUnitServerConfig.builder()
+                .cacheSizeHeapRatio(0.5)
+                .maxCacheSize(100_000)
+                .memoryMode(true)
+                .noSync(true)
+                .build();
+        StreamLog streamLog = new InMemoryStreamLog(new BatchProcessorContext());
+        LogUnitServerCache cache = new LogUnitServerCache(config, streamLog);
+
+        for (int addr = 1; addr < 101; addr++) {
+            LogData logData = generateLogData(addr);
+            streamLog.append(addr, logData);
+        }
+
+        for (int i = 0; i < 500; i++) {
+            cache.get(rnd.nextInt(150));
+        }
```

Comment:
I might have misunderstood something, it seems like there's a bug in the computation of the statistics, the test, or both. I modified the test here to do `cache.get(i)` for `i` between [1, 100] inclusive in order to remove the randomization and have a deterministic result.

Wouldn't we expect the hit ratio to be 1 since the cache contains all of the addresses between [1, 100]? But expected is producing 0 and the test is not failing.

Reply:
I see my misunderstanding. The append doesn't actually populate the cache, so the first get() for an address is a miss.